src/HttpClientFactory/Http/test/DependencyInjection/HttpClientFactoryServiceCollectionExtensionsTest.cs

```diff
@@ -996,5 +1043,35 @@ public TypedClientWithTransientService(HttpClient httpClient, TransientService s
 
             public TransientService Service { get; }
         }
+
+        private class SingleThreadedSynchronizationContext : SynchronizationContext
+        {
+            private readonly Queue<(SendOrPostCallback Callback, object State)> _queue = new Queue<(SendOrPostCallback Callback, object State)>();
```

Comment:
Isn't it somewhat problematic to use `Queue<>` here because it's not thread-safe? (use `ConcurrentQueue<>`) I'd prefer to fix this even if it's not hit in these tests because someone else might move this code or copy from it.

Reply:
Sure, this sync context impl is kinda of a lie anyways. The test may not be testing the right thing TBH, the "fix" is basically that it doesn't queue any work to the sync context outside of top level code.

Reply:
Sure. I'm going to merge this PR and then fix it. I'm worried about someone copying this and creating flaky tests.

Reply:
It looks like this is really a "deadlock detection" sync context, a bit [like this](https://github.com/ramondeklein/deadlockdetection/blob/master/DeadlockDetection/DeadlockDetectionSynchronizationContext.cs). I agree it doesn't feel right for a unit test.